sarracen/kernels/base_kernel.py

```diff
@@ -0,0 +1,53 @@
+class BaseKernel:
+    """A generic kernel used for data interpolation."""
+    def __init__(self, radkernel, cnormk1d, cnormk2d, cnormk3d, wfunc):
+        self._radkernel = radkernel
+        self._cnormk1d = cnormk1d
+        self._cnormk2d = cnormk2d
+        self._cnormk3d = cnormk3d
+        self._wfunc = wfunc
+
+    def w(self, q, dimension):
+        """
```

Comment:
Actually a couple changes to suggest on how kernels are instantiated. Realistically, data is not going to change dimensions. So we don't need to call w() with the dimension every time. Let's change so the dimension is set in the constructor.

def __init__(self, radkernel, ndims, cnormk, wfunc)

def w(self, q)

Will need to refactor the constructors of each kernel to accept number of dimensions and choose the normalization constant based on that. Will also have to refactor the tests accordingly.

Reply:
Changed 👍 I also like how this removes the repetition of cnormk1d, cnormk2d, cnormk3d in the BaseKernel class